src/repl/scala/tools/nsc/interpreter/ReplProps.scala

```diff
@@ -32,6 +32,12 @@ class ReplProps {
   val powerInitCode   = Prop[JFile]("scala.repl.power.initcode")
   val powerBanner     = Prop[JFile]("scala.repl.power.banner")
 
+  val prompt = Prop[String]("scala.shell.prompt")
```

Comment:
Excellent question. Maybe I do not see it correctly, I think, prompt should be rather in Repl than in Properties. And as Properties is object, injecting repl instance is not straitforward.